api/oc_ri.c

```diff
@@ -473,8 +473,79 @@ oc_ri_get_app_resource_by_uri(const char *uri, size_t uri_len, size_t device)
 #endif /* OC_COLLECTIONS */
   return NULL;
 }
+
 #endif /* OC_SERVER */
 
+#ifdef OC_HAS_FEATURE_BRIDGE
+oc_resource_t *
+oc_ri_get_app_resource_by_device(size_t device, bool reset)
+{
+  static oc_resource_t *rsc;
```

Comment:
I'd copy the semantics of `g_query_iterator`, so that we are consistent.

```C

typedef struct {
  oc_resource_t *rsc;
#ifdef OC_COLLECTIONS
  oc_resource_t *coll_rsc;
#endif
} oc_ri_app_resource_iterator_t;

static oc_ri_app_resource_iterator_t g_app_resource_iterator;

void oc_ri_app_resource_iterator_init() {
  g_app_resource_iterator.rsc = oc_ri_get_app_resources();
#ifdef OC_COLLECTIONS
  g_app_resource_iterator.coll_rsc = oc_collection_get_collections();
#endif
}

oc_resource_t* oc_ri_app_resource_iterator_next_for_device(size_t device) {
  ...
}

```

We also need to mention that deleting / adding a resource (collection or non-collection) invalidates the iterator.